### ISSUE TITLE
Implement Trainer

### ISSUE THREAD

**Issue by M-Korniak** [MichalMaszkowski/ZPP#16]
In this task you should implement a training loop in which we train our model. The loop should look more or less like the one we had at DNN classes. 
- data of shape (B,S,C,H,W)
- Should be in module trainer in class Trainer.
- Training should resemble that of a transformer, but the loss should be a reconstruction loss.
- Add training visualisation and printing of train, test loss.
- If data won't be ready before the task - mock the data.
- Use Dataset and Dataloader from Pytorch
- Make sure it works on cpu and gpu.
- the task may require some knowledge about the model in the repo, so ask if help is needed

NOTE: Task is rather complex, so I add myself as an assignee. I may work on this task if I have time.

**Comment by M-Korniak** [comment]
Added a simple training loop as a sketch. Feel free to add some improvements.
